envoy/http/async_client.h

```diff
@@ -24,6 +24,28 @@ using FilterConfigSharedPtr = std::shared_ptr<FilterConfig>;
 } // namespace Router
 namespace Http {
 
+class SidestreamWatermarkCallbacks {
+public:
+  virtual ~SidestreamWatermarkCallbacks() = default;
+
+  /**
+   * Called when the sidestream connection or stream goes over its high watermark. Note that this
+   * may be called separately for both the stream going over and the connection going over. It
+   * is the responsibility of the sidestreamWatermarkCallbacks implementation to handle unwinding
+   * multiple high and low watermark calls.
+   */
+  virtual void onAboveWriteBufferHighWatermark() PURE;
```

Comment:
can we make these sidestream-specific? one of my regrets is how hard it is to tell what call goes where for watermarks as they use such similar names

onSidestreamAboveHighWatermark would be great :-)